shared/teams/delete-team/index.tsx

```diff
@@ -60,16 +60,16 @@ const ReallyDeleteTeam = (props: Props) => {
   const {checkChats, checkFolder, checkNotify} = checks
   const onCheck = (which: keyof typeof checks) => (enable: boolean) => setChecks({...checks, [which]: enable})
   const disabled = !checkChats || !checkFolder || !checkNotify
-  const {deleteWaiting, onBack} = props
+  const {deleteWaiting, onBack, clearWaiting} = props
   const error = Container.useAnyErrors(Constants.deleteTeamWaitingKey(props.teamname))
   const prevDeleteWaiting = Container.usePrevious(deleteWaiting)
   React.useEffect(() => {
     if (prevDeleteWaiting !== undefined && !deleteWaiting && prevDeleteWaiting && !error) {
       // Finished, nav up
       onBack()
     }
-  }, [deleteWaiting, prevDeleteWaiting, onBack])
-  React.useEffect(() => props.clearWaiting, [])
+  }, [deleteWaiting, prevDeleteWaiting, onBack, error])
+  React.useEffect(() => clearWaiting(), [clearWaiting])
```

Comment:
@buoyad please validate, note this wasn't actually calling the function so i assume this does nothing on master

Reply:
This is meant to be called on unmount only, I think the previous is correct?

Reply:
i think this is very tricky, i'd advocate
`() => () => clearWaiting()` maybe so its more clear

Reply:
or () => { return clearWaiting}

Reply:
+1 `() => () => clearWaiting()`

Reply:
Ok, any of those sound fine. But it can't go in as it is currently.